docs: align readme with model (#5348)
canonical/snapcraft: README.md

<img src="https://dashboard.snapcraft.io/site_media/appmedia/2018/04/Snapcraft-logo-bird.png" alt="Snapcraft logo" style="height: 128px; display: block">

# Snapcraft

[![snapcraft](https://snapcraft.io/snapcraft/badge.svg)](https://snapcraft.io/snapcraft)
[![Documentation
Status](https://readthedocs.com/projects/canonical-snapcraft/badge/?version=latest)](https://canonical-snapcraft.readthedocs-hosted.com/en/latest/?badge=latest)
[![Scheduled spread
tests](https://github.com/canonical/snapcraft/actions/workflows/spread-scheduled.yaml/badge.svg?branch=main)](https://github.com/canonical/snapcraft/actions/workflows/spread-scheduled.yaml)

> [![Coverage Status][codecov-image]][codecov-url] [![Code style:
> ruff](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/astral-sh/ruff/main/assets/badge/v2.json)](https://github.com/astral-sh/ruff)

Reviewer: ```suggestion
[![Coverage Status][codecov-image]][codecov-url]
[![Code style:
ruff](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/astral-sh/ruff/main/assets/badge/v2.json)](https://github.com/astral-sh/ruff)
```
Bit of a nitpick, but it's more consistent

Reviewer: Out of curiosity, is there any rule of thumb against just putting every big link in a reference at the bottom? We're already doing it for the codecov stuff and I think it looks nicer, but I can also see others thinking it looks silly since we only use the links once.

Author: > putting every big link in a reference at the bottom

I prefer not to for read
 
 
 
 
 
 
 
 
 
 
 
 
ability.

But I admit its benefit would be clearer if we weren't forced to wrap lines.

Author: For these more functional, non-doc links, I'd be in favour of moving them to the bottom.

Author: All clean: [ebebd38](https://github.com/canonical/snapcraft/pull/5348/commits/ebebd38192686e42272c104abf8294af661a5f58)

**Snapcraft** is the command-line build tool for packaging and distributing software and
apps in the snap container format. It solves the problems of dependency management and
architecture support by bundling all of a software's libraries into the container
itself, and gives developers a way to package any app, program, toolkit, or library for
all major Linux distributions and IoT devices.

## Basic usage

A snap's build configuration is stored in simple language as a project file called
`snapcraft.yaml`, making it easy to add as a new package format to your existing code
base.

From the root of the code base of any software project, Snapcraft creates a minimal
`snapcraft.yaml` with:

```bash
snapcraft init
```

After you add all your project's build and runtime details to the project file, bundle
your project into a snap with:

```bash

> snapcraft

Reviewer 2: Forgot to ask this in review, but is this preferred over `snapcraft pack`?

Reviewer 3: `snapcraft pack` is preferred but it currently doesn't work on `core20` and older.  We're planning to fix that, then eventually deprecate and remove just `snapcraft`.

Reviewer 3: Since this example uses `snapcraft init` above, it will create a `core24` snap.  So recommending `snapcraft pack` works here.

Author: Oof, I had pack in the draft but not in the code. [02540ee](https://github.com/canonical/snapcraft/pull/5348/commits/02540eef3d45202233625437e3993311e1ca01c7)

```

Your project can be registered on public and private app stores, including the Snap
Store:

```bash
snapcraft register
```

Snap versions and revisions, including parallel releases, are published to the store
with:

```bash
snapcraft upload
```

If you're interested in learning more about the Snapcraft commands and how to compose a
project file, try [creating your first
snap](https://snapcraft.io/docs/create-a-new-snap).

## Installation

Snapcraft is available on all major Linux distributions, Windows, and macOS.

Snapcraft itself has first-class support as a snap. On snap-ready systems, you can install it on the command line with:

```bash
sudo snap install snapcraft --classic
```

For complete installation, you need an additional Linux container tool. Snapcraft can
also be installed as a traditional package on many popular Linux repositories. For help
with both, we cover how to [set up

> Snapcraft](https://canonical-snapcraft.readthedocs-hosted.com/en/stable/howto/set-up-snapcraft)

Author: Idiomatic example of a line we can't keep within an 88 char limit. :upside_down_face:

> also be installed as a traditional package on many popular Linux repositories. For help
> with both, we cover how to [set up
> Snapcraft](https://canonical-snapcraft.readthedocs-hosted.com/en/stable/howto/set-up-snapcraft)

Reviewer 2: I interpret "for help with both" as meaning "to get help with these things", so it feels weird to follow that up by stating something that we do.

```suggestion
also be installed as a traditional package on many popular Linux repositories. For help
with both, read our documentation on how to [set up
Snapcraft](https://canonical-snapcraft.readthedocs-hosted.com/en/stable/howto/set-up-snapcraft)
```

Author: I'd like to avoid the imperative here. How about:

> If you need help with either, the documentation covers how to set up Snapcraft.

Author: 38d910e

in the docs.

## Documentation

The Snapcraft docs provide guidance and learning material about the full process of
building a project file, debugging snaps, resolving interfaces, the command reference,
and much more:

- [Snapcraft build guide on snapcraft.io](https://snapcraft.io/docs)
- [Snapcraft documentation on
  ReadTheDocs](https://canonical-snapcraft.readthedocs-hosted.com/en/stable/)

## Community and support

We are a growing community of crafters who build snaps for all Linux software.

Ask your questions about Snapcraft and what's on the horizon, and see who's working on
what in the [Snapcraft Forum](https://forum.snapcraft.io) and on the [Snapcraft Matrix
channel](https://matrix.to/#/#snapcraft:ubuntu.com).

You can report any issues or bugs you find on
[GitHub](https://github.com/canonical/snapcraft/issues) or

> [Launchpad](https://bugs.launchpad.net/snapcraft/+filebug).

Author: Do we want to keep the Launchpad link?

Reviewer: I'm for removing it, we generally prefer bug reports here, but @mr-cal should give a final call.

Reviewer 3: Yes please remove it!

Author: 38d910e
